ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/ccd/services/CoreCaseDataService.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.hmcts.cmc.ccd.migration.ccd.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.migration.ccd.repositories.CCDCaseApi;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.reform.ccd.client.model.EventRequestData;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.cmc.ccd.migration.ccd.repositories.CCDCaseApi.CASE_TYPE_ID;
+import static uk.gov.hmcts.cmc.ccd.migration.ccd.repositories.CCDCaseApi.JURISDICTION_ID;
+
+@Service
+@ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
+public class CoreCaseDataService {
+
+    enum EventType {
+        MIGRATED_FROM_CLAIMSTORE("migratedFromClaimstore"),
+        SUBMIT_CLAIM("submitClaimEvent");
+
+        private String value;
+
+        EventType(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+    }
+
+    private final MigrateCoreCaseDataService migrateCoreCaseDataService;
+    private final CCDCaseApi ccdCaseApi;
+
+    @Autowired
+    public CoreCaseDataService(
+        MigrateCoreCaseDataService migrateCoreCaseDataService,
+        CCDCaseApi ccdCaseApi
+    ) {
+        this.migrateCoreCaseDataService = migrateCoreCaseDataService;
+        this.ccdCaseApi = ccdCaseApi;
+    }
+
+    public void create(String authorisation, Claim claim) {
+        try {
+            EventRequestData eventRequestData = EventRequestData.builder()
+                .userId(claim.getSubmitterId())
+                .jurisdictionId(JURISDICTION_ID)
+                .caseTypeId(CASE_TYPE_ID)
+                .eventId(EventType.SUBMIT_CLAIM.getValue())
+                .ignoreWarning(true)
+                .build();
+
+            migrateCoreCaseDataService.save(
+                authorisation,
+                eventRequestData,
+                claim
+            );
+        } catch (Exception exception) {
+            throw new RuntimeException(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS00112)

Reply:
ok